airbyte-cdk/bulk/toolkits/load-object-storage/src/main/kotlin/io/airbyte/cdk/load/pipline/object_storage/ObjectLoaderPipeline.kt

```diff
@@ -63,4 +64,36 @@ class ObjectLoaderPipelineWithFiles<K : WithStream, T : RemoteObject<*>>(
     recordPartStep,
     recordUploadStep,
     recordCompleterStep,
-))
+)) {
+    companion object {
+        fun hasFileTransfer(catalog: DestinationCatalog): Boolean =
+            catalog.streams.any { it.includeFiles }
+
+        fun <K : WithStream, T : RemoteObject<*>> selectPipelineSteps(
+            catalog: DestinationCatalog,
+            routeEventStep: RouteEventStep,
+            fileChunkStep: FileChunkStep<T>,
+            @Named("filePartLoaderStep") fileChunkUploader: ObjectLoaderPartLoaderStep<T>,
+            @Named("fileUploadCompleterStep") fileCompleterStep: ObjectLoaderUploadCompleterStep<K, T>,
+            forwardFileRecordStep: ForwardFileRecordStep<T>,
+            @Named("fileRecordPartFormatterStep") recordPartStep: ObjectLoaderPartFormatterStep,
+            @Named("recordPartLoaderStep") recordUploadStep: ObjectLoaderPartLoaderStep<T>,
+            @Named("recordUploadCompleterStep") recordCompleterStep: ObjectLoaderUploadCompleterStep<K, T>,
```

Comment:
Nit: We don't need the named annotations here since this isn't getting injected by micronaut.

Reply:
Also, I'd probably just inline this, but don't feel too strongly. Is the thought to be able to unit test this?

Reply:
Removed the annotations. I was wondering if we wanted to inline this, at the same time, having a 10line conditional in the constructor didn't feel like the most readable option.